pkg/storage/tsdb/bucketindex/updater.go

```diff
@@ -74,13 +74,19 @@ func (w *Updater) UpdateIndex(ctx context.Context, old *Index) (*Index, map[ulid
 	}, partials, totalBlocksBlocksMarkedForNoCompaction, nil
 }
 
-func (w *Updater) updateBlocks(ctx context.Context, old []*Block) (blocks []*Block, partials map[ulid.ULID]error, _ error) {
+func (w *Updater) updateBlocks(ctx context.Context, old []*Block, deletedBlocks map[ulid.ULID]struct{}) (blocks []*Block, partials map[ulid.ULID]error, _ error) {
 	discovered := map[ulid.ULID]struct{}{}
 	partials = map[ulid.ULID]error{}
 
 	// Find all blocks in the storage.
 	err := w.bkt.Iter(ctx, "", func(name string) error {
 		if id, ok := block.IsBlockDir(name); ok {
+
+			if _, ok := deletedBlocks[id]; ok {
+				level.Warn(w.logger).Log("msg", "skipped block with missing global deletion marker", "block", id.String())
```

Comment:
I am wondering if this log line is necessary or not. This is the case where we have the dangling visit-mark for the block only? We skip it anyway so to me no need to log anything

Reply:
That is true. This will also include partial blocks. The point of these log was to get specif scenarios which are not suppose to happen as a block going back to active.
Looking at the code i can move this check for the existent block loop. That will ignore any discovered block.